Specify how errors are formatted (#34)
peppelinux/draft-demarco-acme-openid-federation: draft-demarco-acme-openid-federation.md

The X.509 Certificate revocation request is defined in the [Section
7.6](https://datatracker.ietf.org/doc/html/rfc8555#section-7.6) of [RFC8555].

# Errors

This document defines one new error type URI to be used in problem documents
{{!RFC9457}}, as described in {{Section 6.7 of !RFC8555}}.

The error type `urn:ietf:params:ppm:acme-openid-federation:error:oauthError` can
be used to encapsulate any OAuth error code returned while resolving OpenID
Federation entities. The title of this error type is "OAuth Error". The `detail`
member of the document MAY include the description of the particular OAuth error

> code that caused the error. This error type has an extension member named
> `error_code`. The `error_code` member SHOULD be set to the OAuth error code,

Reviewer: Let's be clear about where this extension member goes

```suggestion
code that caused the error. The problem document for this error type SHOULD include an extension
member named `error_code`. The `error_code` member SHOULD be set to the OAuth error code,
```

The X.509 Certificate revocation request is defined in the [Section
7.6](https://datatracker.ietf.org/doc/html/rfc8555#section-7.6) of [RFC8555].

# Errors

This document defines one new error type URI to be used in problem documents
{{!RFC9457}}, as described in {{Section 6.7 of !RFC8555}}.

The error type `urn:ietf:params:acme:error:openIDFederationEntity` can
be used to encapsulate any OAuth error code returned while resolving OpenID

> Federation entities. The title of this error type is "OpenID Federation Error".

Reviewer 2: ```suggestion
Federation Entities. The title of this error type is "OpenID Federation Error".
```

([1.3.6.1.5.5.7.8](https://www.iana.org/assignments/smi-numbers/smi-numbers.xhtml#smi-numbers-1.3.6.1.5.5.7.8))
and reference this document.

## URN Sub-namespace for ACME with OpenID Federation (urn:ietf:params:acme-openid-federation) {#urn-namespace}

> IANA is asked to register the following value in the "IETF URN Sub-namespace for

Reviewer: I am missing why this can't just go as another entry in [ACME Error Types](https://www.iana.org/assignments/acme/acme.xhtml#acme-error-types). We need to justify creasing a new registry, particularly for a single value, and I don't see that yet.

Author: Good point, will do.

([1.3.6.1.5.5.7.8](https://www.iana.org/assignments/smi-numbers/smi-numbers.xhtml#smi-numbers-1.3.6.1.5.5.7.8))
and reference this document.

## ACME Error Types

IANA is asked to add the following assignment to the "ACME Error Types"
registry, in the "Automated Certificate Management Environment (ACME) Protocol"
registry group {{IANA-ACME}}.

|Type|Description|Reference|
|----|-----------|---------|

> |openIDFederationError|An error occurred while resolving an OpenID Federation entity|RFC XXXX|

Reviewer: Since `error` is already in the URN, and [none of the other error types include `error`](https://www.iana.org/assignments/acme/acme.xhtml#acme-error-types), what do you think of this being `openIDFederationEntity`?

```suggestion
|openIDFederationEntity|An error occurred while resolving an OpenID Federation entity|RFC XXXX|
```
